### PR TITLE
Support Depth endpoint in SpotApi

### PR DESCRIPTION
**Pull request by Swoorup**
OrderBook Depth support as in https://binance-docs.github.io/apidocs/spot/en/#order-book

### CODE REVIEW

File: src/main/scala/io/github/paoloboni/binance/spot/SpotApi.scala [hunk 1]
```diff
@@ -50,6 +50,24 @@ final case class SpotApi[F[_]: Logger](
...
+    * @return
+    *   the orderbook depth
+    */
+  def getDepth(query: common.parameters.Depth): F[Depth] =
+    for {
+      uri <- F.fromEither(Try(uri"${config.restBaseUrl}/api/v3/depth").map(_.addParams(query.toQueryParams)).toEither)
+      depthOrError <- client.get[CirceResponse[Depth]](
+        uri = uri,
+        responseAs = asJson[Depth],
+        limiters = rateLimiters.filterNot(_.limitType == common.response.RateLimitType.ORDERS),
```
Comment (paoloboni) [inline src/main/scala/io/github/paoloboni/binance/spot/SpotApi.scala:66]:
```suggestion
        limiters = rateLimiters.filterNot(_.limitType == common.response.RateLimitType.ORDERS),
        weight = query.limit.weight // need to set the weight based on the limit parameter
```

File: src/main/scala/io/github/paoloboni/binance/common/parameters/Depth.scala [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package io.github.paoloboni.binance.common.parameters
+
+final case class Depth(symbol: String, limit: Int)
```
Comment (paoloboni) [inline src/main/scala/io/github/paoloboni/binance/common/parameters/Depth.scala:24]: Based on the API spec, the limit could be modelled as enum for both version 2.x and 3.x of Scala.
Specs:

Limit | Weight
-- | --
5, 10, 20, 50, 100 | 1
500 | 5
1000 | 10
5000 | 50

Scala 3 (under scala-3 directory):
```
enum DepthLimit(val weight: Int) derives EnumDecoder:
  case 5 extends DepthLimit(1)
  ...
  case 500 extends DepthLimit(5)
  ...
```
Scala 2 (under scala-2 directory):
```
sealed abstract class DepthLimit(val weight: Int) extends EnumEntry
object Interval extends Enum[DepthLimit] with CirceEnum[DepthLimit] {
  val values: immutable.IndexedSeq[DepthLimit] = findValues

  case object `5` extends DepthLimit(1)
  ...
  case object `500` extends DepthLimit(5)
  ...
}
```

**Review by paoloboni — COMMENTED** [review]
Looks good, but the rate limit is not applied as specified by the API specs, this can cause the client to exceed the rate limit.

File: src/main/scala/io/github/paoloboni/binance/common/parameters/Depth.scala (hunk 2 above)
Comment (Swoorup) [reply src/main/scala/io/github/paoloboni/binance/common/parameters/Depth.scala]: @paoloboni Done.

**Comment by paoloboni** [comment]
@all-contributors please add @Swoorup for coding

**Pull request merged**